kbfsgit/fs_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package kbfsgit
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/keybase/kbfs/libfs"
+	"github.com/keybase/kbfs/libkbfs"
+	"github.com/keybase/kbfs/tlf"
+	"github.com/stretchr/testify/require"
+	gogit "gopkg.in/src-d/go-git.v4"
+	gogitcfg "gopkg.in/src-d/go-git.v4/config"
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
+	"gopkg.in/src-d/go-git.v4/storage"
+	"gopkg.in/src-d/go-git.v4/storage/filesystem"
+)
+
+func makeFS(t *testing.T, subdir string) (
+	context.Context, *libkbfs.TlfHandle, *libfs.FS) {
+	ctx := libkbfs.BackgroundContextWithCancellationDelayer()
+	config := libkbfs.MakeTestConfigOrBust(t, "user1", "user2")
+	h, err := libkbfs.ParseTlfHandle(ctx, config.KBPKI(), "user1", tlf.Private)
+	require.NoError(t, err)
+	fs, err := libfs.NewFS(ctx, config, h, subdir, "")
+	require.NoError(t, err)
+	return ctx, h, fs
+}
+
+// configInMemoryStorer keeps the git config in memory, to work around
+// a gcfg bug (used by go-git when reading configs from disk) that
+// causes a freakout when it sees backslashes in git file URLs.
```

Comment:
I see; Thanks for the explanation! I still don't quiet understand the need for a remote in the bare repo though (also asked [here](https://github.com/keybase/kbfs/pull/1120#discussion_r134866980))